### DISCUSSION TITLE
How did you 'globalThis'ify Vanta Waves?

### DISCUSSION THREAD

**Discussion by rymix** [DustinBrett/daedalOS#245] [Q&A] [answered]
Could you please explain how you were able to generate this version of Vanta Waves that used globalThis over window?

https://github.com/DustinBrett/daedalOS/blob/7c342176917b1664a5c388ed6c2bbb41a29061f9/public/System/Vanta.js/vanta.waves.min.js

I would like to convert some other Vanta animated backgrounds in a similar manner, but I do not relish the prospect of working my way through a minified javascript file.

Did you recompile using a modified version of the open source project here? https://github.com/tengbao/vanta

**Comment by DustinBrett** [comment, marked as answer]
Thanks for the interest. I did indeed convert from source using the vanta repo. I had PR'd (https://github.com/tengbao/vanta/pull/118) one of my tweaks to the repo. I think I mostly just did a global search/replace for `window` to `globalThis`. I run it in a web worker so this was required. Attached is a diff of all the changes I've done to my local fork of vanta.

[vanta-ea669eb-Various tweaks.zip](https://github.com/DustinBrett/daedalOS/files/10198439/vanta-ea669eb-Various.tweaks.zip)

**Comment by rymix** [comment]
Thank you! Very helpful. Vanta Clouds is now globalThisified. I'll move onto the rest shortly.
![CloudsGlobalThisified](https://user-images.githubusercontent.com/75093032/206862119-71eba36c-e8dd-4964-ae36-12feaff23928.png)

**Comment by DustinBrett** [comment]
Very cool! I like that cloud one but on mobile for me it didn't look too good which is why I hadn't integrated a "cloud" one yet. Maybe I'll need to reconsider as it's quite cool.
